rpc/src/eth_rpc.rs

```diff
@@ -180,6 +193,37 @@ impl EthApi {
         Ok(header_with_proof.header)
     }
 
+    async fn fetch_block(
+        &self,
+        header: Header,
+        hydrated_transactions: bool,
+    ) -> Result<Block, RpcServeError> {
+        if hydrated_transactions {
+            return Err(RpcServeError::Message(
+                "replying with all transaction bodies is not supported yet".to_string(),
+            ));
+        }
+
+        let body = self.fetch_block_body(header.hash()).await?;
+        let transactions = match &body {
+            BlockBody::Legacy(body) => &body.txs,
+            BlockBody::Merge(body) => &body.txs,
+            BlockBody::Shanghai(body) => &body.txs,
+        };
```

Comment:
I just wanted to mention body has a getter for transactions, I think we could remove the Result and clones from the getter, but not a big deal either way